### PR TITLE
Post Filtering, Profile Updates, and API Enhancements

### PR DESCRIPTION
**Pull request by Mutti499**
This PR includes:
- Post filtering by status, tags, languages, date, and sorting by popularity.
- Profile updates: view bookmarks, comments, and annotations.
- API improvements: /mark_as_unanswered and enhanced /get_profile endpoint.
- Users can upvote posts, search profiles, and explore questions without registration.
- Includes additional minor fixes and documentation updates.
Both the web frontend and backend should test the features.
@ceylingebes @serhan-cakmak

### CODE REVIEW

**Comment by serhan-cakmak** [comment]
Great job mate! Yet I would like you to check out the following points:

1- Currently, there is an inconsistency in the programming languages field of the questions. Some questions were posted when we got the language name without the version, yet this approach caused problems while trying to fetch the languages during code execution. Thus,  we modified this field to also include the version of the language. Having said that, I want to point out that we need to update the old questions' programming language field. Also we need to update the language selection in the front end accordingly. 

old question and filter 
image


new version
image

2- Users can see their annotations, but the annotation object doesn't make any sense in the current version; it should be able to redirect the user to the page where the annotation is made. 

3- "Users can upvote posts, search profiles, and explore questions without registration." can cause many problems both in the front end and database if we don't implement the functionalities carefully. Considering the time left, I am not sure if we should implement this functionality since it doesn't enhance the UX a lot.

**Comment by Mutti499** [comment]
1- You are right and data should be updated from database ASAP.
2 - Frontend's job. I am only giving some template to make it visible.
3 - Okey mate lets posptpone that and focus on important tasks!

**Review by halil-karabacak — APPROVED** [review]

**Comment by ErayEroglu** [comment]
I am taking the reviewing responsibility, I will test the features and let you know if there exists any issues. It seems like you've done a lot of things @Mutti499 , even reviewing will take some time. I can't imagine the effort you spent

**Comment by ErayEroglu** [comment]
I tried filtering with Python language, and without specifying any tag, but I returned nothing. When I manually checked the questions, I saw there exists such questions. Can you check @Mutti499, I can help to bug fix if you wish

**Comment by ErayEroglu** [comment]
When I entered my profile page, bookmarks are listed there but userID's are not seen. It would be better if we check

**Comment by ErayEroglu** [comment]
Not a feature you implemented, but I think it would be better if I could navigate to the question from the comments section on the profile page. I am opening an issue for this to frontend guys, it would be great to have

**Comment by halil-karabacak** [comment]
> I tried filtering with Python language, and without specifying any tag, but it returned nothing. When I manually checked the questions, I saw there exists such questions. Can you check @Mutti499, I can help to bug fix if you wish

Amazing catch!

File: koduyorum-web/src/Feed.css [hunk 1]
```diff
@@ -231,11 +369,15 @@ body {
...
-      margin-bottom: 1rem;
+    .contributors-container,
+    .filters {
+        width: 100%;
+        margin-bottom: 1rem;
     }
-  
+
+    .filters {
+      order: -1; /* Makes filters appear at the top on mobile */
```
[the remark this answers is not in the document]
Comment (Mutti499) [reply koduyorum-web/src/Feed.css:379]: Saul

**Comment by Mutti499** [comment]
The problem mentıoned here are solved ın the branch swagger so lets proceed and merge these two branches.

**Comment by ErayEroglu** [comment]
I am double checking it now, unless there is something wrong I will merge it

**Comment by ErayEroglu** [comment]
Eline sağlık @Mutti499, everything is working as expected. I am merging to main. Keep up the good work!

**Pull request merged**